scripts/source-maintenance.sh

```diff
@@ -401,16 +447,26 @@ printAmFile TRANSLATE_LANGUAGES "doc/manuals/" "*.po" | sed 's%\.po%\.lang%g' >
 printAmFile STUB_SOURCE "src/" "tests/stub_*.cc" > src/tests/Stub.am
 
 # Build the GPERF generated content
-make -C src/http gperf-files
+if [ -f src/http/Makefile ]; then
+    make -C src/http gperf-files
+else
+    echo "source tree not configured; skipping regeneration of xperf sources"
```

Comment:
> Regenerating xperf files is a rare occurrence, isn't it? I've iterated by skipping the step and emitting a warning instead

This change is a step backwards IMO: The official code already violates the principles I outlined [earlier](https://github.com/squid-cache/squid/pull/830#discussion_r643953882) -- it ignores gperf generation failures (with an error message but that is a minor detail). The change replaces the error message with a better one, _but_ it also codifies that gperf failures are OK to ignore. This is not longer an old bug in the neglected script area, but a fresh Project decision -- we are OK with gperf failing. I recommend undoing this change until we are ready to support this properly (in a dedicated PR, as discussed earlier).

If you insist on this change (not sure _why_ you would -- the ultimate outcome of this code does not really change, only the error message, AFAICT!), then I will add source comments with XXXs to document that what this code is doing now is still _wrong_.

Reply:
okay, done